### ISSUE TITLE
Code diagnostics for JavaScript doesn't understand computed properties

### ISSUE THREAD

**Issue by Selbosh** [rstudio/rstudio#2009]
Code diagnostics for JavaScript doesn't like [computed properties](http://javascript.info/object#computed-properties) in RStudio 1.1.383 on Windows.

### MWE

This is valid code and runs fine in Chrome.

```javascript
"use strict";

let property = "bar";
let foo = {
  [property]: 42
};

console.log(foo.bar);
```

### Error in linter

But RStudio's code diagnostics throw this up with an :x: next to line 5.

```
Expected '}' to match '{' from line 4 and instead saw '['.
Missing ";" before statement
Missing semicolon
Expected an identifier and instead saw ']'.
Expected '(end)' and instead saw ':'.
```

I looked to see if there was a way of user-customising the RStudio linter but I couldn't find anything in the documentation, so have no choice but to file this as a bug.

**Comment by kevinushey** [comment]
Thanks for the bug report! This particular linter is packaged as part of the JavaScript mode provided to us by Ace, which we use as our source editor. It appears that this issue is fixed in the development version of Ace; hopefully we'll be able to pick up that fix when we next upgrade Ace.

**Comment by jmcphers** [comment]
This is fixed in the new version of Ace included in RStudio 1.3. If you want to try it:

https://dailies.rstudio.com/
